src/button.ts

```diff
@@ -1,35 +1,34 @@
-import {Sprite, Text} from "pixi.js";
+import {ITextStyle, Sprite, Text, Texture} from "pixi.js";
 
 interface Options {
     label?: {
-        text: string,
-        style: object
+        text?: string,
+        style?: Partial<ITextStyle>
     };
-    textures?: { [index: string]: any };
-
+    textures?: { [key: string]: Texture };
 }
 
 export default class Button extends Sprite {
     private _options: Options;
     private _label: Text;
     private _isPressed: boolean;
     private _isHover: boolean;
-    private _isEnabled: boolean;
 
     buttonMode: boolean;
     interactive: boolean;
 
-    constructor(options: object) {
+    constructor(options: Options) {
         super();
         this._options = options;
         this._init();
     }
 
     private _init() {
         this.buttonMode = true;
+        this.interactive = true;
```

Comment:
there's already a dedicated method for
```
this.interactive = true;
this._updateTexture();
```
`this.enable()`
This is another pattern to learn - DRY (Don't Repeat Yourself)